subprojects/docs/src/docs/userguide/building_cpp_projects.adoc

```diff
@@ -0,0 +1,275 @@
+// Copyright 2018 the original author or authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[building_cpp_projects]]
+= Building C++ projects
+
+Gradle uses a convention-over-configuration approach to building native projects.
+If you are coming from another native build system, those concepts may be overwhelming at first but serve the purpose to simplify the whole build script authoring.
+It borrows several conventions from Apache Maven.
+In particular, it uses the same default directory pattern for source files and headers.
+
+We will look at C\++ projects in detail in this chapter, but most of the topics apply to other supported native languages as well.
+If you don’t have much experience with building native projects with Gradle, take a look at the C++ tutorials for step-by-step instructions on how to build various types of basic C++ projects as well as some common use case.
+
+The C++ plugins covered in this chapter were https://blog.gradle.org/introducing-the-new-cpp-plugins[introduced in 2018] and we recommend users to use those plugins over <<native_software.adoc#native_binaries,the older Native plugins>>.
+
+== Introduction
+
+The simplest build script for a C\++ project applies the C++ application plugin or the C++ library plugin and optionally sets the project version:
```

Comment:
That is a very good question, it seems that sometimes the `++` is being swallowed by Asciidoc and time not. I just escaped the one that was swallowed so they display properly.

Reply:
Asciidoctor has an attribute for this: `{cpp}`. I recommend using it rather than escaping some of the pluses. I think the current solution is not robust to changes in the text.

Reply:
Great idea Peter, I will change it everywhere.